### ISSUE TITLE
Buffer size

### ISSUE THREAD

**Issue by ygoe** [dotnet/docs#5108]
How large must the buffer array be? What happens if it was too small, so the received message was longer than the array? Can I just come back with more space to receive the rest? I guess the result.EndOfMessage property should be checked to determine that, but that's just a guess.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9f712007-895c-803c-b566-9ef3a8270834
* Version Independent ID: 98b29b49-2e42-4a20-330f-ee1546517b01
* Content: [ClientWebSocket.ReceiveAsync(ArraySegment<Byte>, CancellationToken) Method (System.Net.WebSockets)](https://docs.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocket.receiveasync?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(System.Net.WebSockets.ClientWebSocket.ReceiveAsync);k(DevLang-csharp)%26rd%3Dtrue&view=netframework-4.7.2)
* Content Source: [xml/System.Net.WebSockets/ClientWebSocket.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Net.WebSockets/ClientWebSocket.xml)
* Service: **unspecified**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by karelz** [comment]
@caesar1995 can you please help answer the question? I wonder if the behavior is the same on all platforms we support.

**Comment by caesar1995** [comment]
> I guess the result.EndOfMessage property should be checked to determine that, but that's just a guess.

Correct. We used to have issue with `result.EndOfMessage` on .NET Core, and we have fixed it: https://github.com/dotnet/corefx/pull/28325

As for the sample code of a receive loop, please refer to the test case here: 
https://github.com/dotnet/corefx/pull/28325/files#diff-0dc72bcdecaf43bf2632f602e91b6866R135

> I wonder if the behavior is the same on all platforms we support.

Yes. We use managed websocket on Core. And this is the same behavior on Framework.

**Comment by ygoe** [comment]
@caesar1995 Oh there's a bug in .NET Core < 2.1? I don't understand what it's about. At least I'm not having any trouble yet (with .NET Core 2.0).

And the memory management of the test case doesn't look too ideal. Every chunk of bytes is copied twice. I was hoping to reduce that. I'm using a single byte array to write to and resize that when it's too small (similar to what `List<T>` does internally).

**Comment by ygoe** [comment]
I haven't set up any load testing yet. Still in the process of getting the basics complete. I just tried not to waste too many resources from the start. Some may see this as premature optimisation. It depends.
